core/src/main/scala/quasar/fs/Natural.scala

```diff
@@ -21,12 +21,19 @@ final class Natural private (val value: Long) {
 
   def toInt: Int =
     value.toInt
+
+  override def equals(other: scala.Any) = other match {
```

Comment:
Possibly, though there is typelevel/scalaz-specs2 that provides a `must equal` that uses `Equal[_]` instead of `equals`.